aiida/schedulers/datastructures.py

```diff
@@ -169,6 +174,11 @@ def accepts_default_mpiprocs_per_machine(cls):
         """Return True if this subclass accepts a `default_mpiprocs_per_machine` key, False otherwise."""
         return True
 
+    @classmethod
+    def accepts_default_mem_per_machine(cls):
+        """Return True if this subclass accepts a `default_mem_per_machine` key, False otherwise."""
+        return True
```

Comment:
I agree with that. Done.

Reply:
Should we just remove this function entirely? It has always been possible to set `job_tmpl.max_memory_kb`, and I am not changing this part.

Reply:
yes, but `job_tmpl` is an instance of `JobTemplate` which is not the same as the `JobResource` which is what you are adding this method to. So whether the resource accepts it has nothing to do with the job template.

As I understand it, the `accepts_default_mpiprocs_per_machine` method is there primarily such that `verdi computer setup` knows whether it should prompt for this option, based on the scheduler plugin that is specified.

But I think all changes of the `JobResource` you have done should be removed. It is ultimately called by the `CalcJob` that will pass its `metadata.options.resources` input to `validate_resources` of the `JobResource` of the scheduler plugin. The input `memory.options.max_memory_kb` will never be passed into this, so it is pointless to add this code. It will never do anything.

At most, one could think that some scheduler plugins can not do anything with the setting `max_memory_kb`. But then we should change the `Scheduler` interface to allow to specify this, not the job resources.

Reply:
I removed the code, please resolve if you agree.